lib/config/options/index.ts

```diff
@@ -79,6 +79,47 @@ const options: RenovateOptions[] = [
     default: 'update',
     cli: false,
   },
+  {
+    name: 'preUpgradeTasks',
+    description:
+      'Pre-upgrade tasks that are executed before updating dependency files by Renovate.',
+    type: 'object',
+    default: {
+      commands: [],
+      fileFilters: [],
+      executionMode: 'update',
+    },
+  },
+  {
+    name: 'commands',
+    description:
+      'A list of pre-upgrade commands that are executed before updating dependency files by Renovate.',
+    type: 'array',
+    subType: 'string',
+    parent: 'preUpgradeTasks',
```

Comment:
@rarkins I think we need to allow multiple parents, so we need to define commands and others only once.